Automattic/simplenote-macos
# Checklists!

Using a very similar approach (and copied code!) from https://github.com/Automattic/simplenote-ios/pull/258, this PR adds a checklist UI, using markdown syntax behind the scenes.

<img width="225" alt="screen shot 2018-12-17 at 3 33 30 pm" src="https://user-images.githubusercontent.com/789137/50122639-59287000-0212-11e9-8fbf-09445b85f52b.png">

**To Test**
* Open a note, and create a checklist (look in the `Format` menu)! Pressing return at the end of a checklist item should automatically add a new one on the next line.
* Open the note on app.simplenote.com, it should have the correct markdown syntax displayed.
* Update the note on the web browser, the app should update and display any modified checklists appropriately.
* Test things that modify the note content, like the history browser and ensure they work as expected.
